### PR TITLE
Script updates for tooling

### PR DESCRIPTION
**Pull request by willchavez**
- diff.sh, now also downloads the previous versions of the updates file, and uses imagemagick to compare them.
- Generate HTML Report, python script that creates a replica of the visual-diff tool we use for icon updates

icon_updates

### CODE REVIEW

File: scripts/diff.sh [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+# Ensure the script is run from the correct directory
+# Do not delete this check, this scripts needs to be ran at the root of the repo.
+if [[ $(basename "$PWD") == "scripts" ]]; then
+    echo "Please run this script from the root directory of your repository."
+    exit 1
+fi
+
+# Prompt the user for the commit IDs
+read -p "Enter the commit ID from (inclusive): " commit_id_from
+read -p "Enter the commit ID to: " commit_id_to
```
Comment (spencer-nelson) [inline scripts/diff.sh:27]: Is it possible to just run this with a from and no "to" so it runs to the current commit?

File: scripts/diff.sh [hunk 2]
```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+
+# This script extracts files changed between two specified Git commit IDs and performs various file operations.
```
Comment (spencer-nelson) [inline scripts/diff.sh:3]: For this to work you have to be on a branch that has the specific commit ids as part of the branch history. Sometimes it just requires a git pull, sometimes you might have to check out another branch, but that's worth noting as a possible error.

File: scripts/generate_html_report.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+Note:
+- The HTML report includes buttons to toggle the visibility of new icons and to zoom in/out on the icons.
+"""
+
+import os
+import re
+
+# Prompt the user for the path to the SVGs folder
+svg_folder = input("Please enter the path to the SVGs folder: ")
```
Comment (spencer-nelson) [inline scripts/generate_html_report.py:39]: can there be a default value to this so one can just hit enter and not have to add a path?

**Review by spencer-nelson — CHANGES_REQUESTED** [review]
The default values are a nicety, but I feel it's critical to include instructions about being on a valid branch. I've been caught by that too many times.

**Comment by willchavez** [comment]
@spencer-nelson just made some updates, excellent suggestions. 💯

**Review by spencer-nelson — APPROVED** [review]

**Pull request merged**